### ISSUE TITLE
[CI] MvSliceTests testEvaluateBlockWithoutNulls {TestCase=<cartesian_shape>, <integer>, <integer>} failing

### ISSUE THREAD

**Issue by alex-spies** [elastic/elasticsearch#107756]
**Build scan:**
https://gradle-enterprise.elastic.co/s/d7k5gdah5h3z4/tests/:x-pack:plugin:esql:test/org.elasticsearch.xpack.esql.expression.function.scalar.multivalue.MvSliceTests/testEvaluateBlockWithoutNulls%20%7BTestCase=%3Ccartesian_shape%3E,%20%3Cinteger%3E,%20%3Cinteger%3E%7D

**Reproduction line:**
```
./gradlew ':x-pack:plugin:esql:test' --tests "org.elasticsearch.xpack.esql.expression.function.scalar.multivalue.MvSliceTests.testEvaluateBlockWithoutNulls {TestCase=<cartesian_shape>, <integer>, <integer>}" -Dtests.seed=D767D3FC112B3251 -Dtests.configure_test_clusters_with_one_processor=true -Dtests.locale=th-TH -Dtests.timezone=Egypt -Druntime.java=21
```

**Applicable branches:**
main

**Reproduces locally?:**
Didn't try

**Failure history:**
[Failure dashboard for `org.elasticsearch.xpack.esql.expression.function.scalar.multivalue.MvSliceTests#testEvaluateBlockWithoutNulls {TestCase=<cartesian_shape>, <integer>, <integer>}`](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupInput:(chainingSystem:HIERARCHICAL,controlStyle:twoLine,ignoreParentSettings:(ignoreFilters:!f,ignoreQuery:!f,ignoreTimerange:!f,ignoreValidations:!t),panels:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:task.keyword,grow:!t,id:'0c0c9cb8-ccd2-45c6-9b13-96bac4abc542',searchTechnique:wildcard,selectedOptions:!(),singleSelect:!t,title:'Gradle%20Task',width:medium),grow:!t,order:0,type:optionsListControl,width:small),'144933da-5c1b-4257-a969-7f43455a7901':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:name.keyword,grow:!t,id:'144933da-5c1b-4257-a969-7f43455a7901',searchTechnique:wildcard,selectedOptions:!('testEvaluateBlockWithoutNulls%20%7BTestCase%3D%3Ccartesian_shape%3E,%20%3Cinteger%3E,%20%3Cinteger%3E%7D'),title:Test,width:medium),grow:!t,order:2,type:optionsListControl,width:medium),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:className.keyword,grow:!t,id:'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850',searchTechnique:wildcard,selectedOptions:!('org.elasticsearch.xpack.esql.expression.function.scalar.multivalue.MvSliceTests'),title:Suite,width:medium),grow:!t,order:1,type:optionsListControl,width:medium)))))

**Failure excerpt:**
```
java.lang.OutOfMemoryError: Java heap space


```

**Comment by nik9000** [comment]
Fun times. That change probably changed the randomization seed but didn't change the ability to reproduce the test. I can have a look in a bit though.

**Comment by fang-xing-esql** [comment]
This should be resolved by #108195.
